File: tests/parsers/samples/code/rb/complex.rb
module ComplexMetrics
  module_function

  def distance(a, b, scale = 1.0)
    (a - b).abs * scale
  end

  def root_mean_square(values)
    return 0.0 if values.empty?
    total = values.reduce(0.0) { |acc, value| acc + value * value }
    Math.sqrt(total / values.length)
  end
end

File: tests/parsers/samples/code/dart/complex.dart
import 'dart:math';

class ComplexMetrics {
  const ComplexMetrics({this.scale = 1.0});

  final double scale;

  double distance(num a, num b) {
    return (a - b).abs() * scale;
  }

  double rootMeanSquare(List<double> values) {
    if (values.isEmpty) {
      return 0;
    }
    final total = values.map((v) => v * v).reduce((a, b) => a + b);
    return sqrt(total / values.length);
  }

  static ComplexMetrics buildDefault() => const ComplexMetrics(scale: 0.8);
}
